src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs

```diff
@@ -0,0 +1,553 @@
+
+                ((IWsFederationApplicationSettingsApplication)retrievedSettings.App).AudienceRestriction.Should().Be("urn:example:app");
+                ((IWsFederationApplicationSettingsApplication)retrievedSettings.App).GroupName.Should().BeNullOrEmpty();
+                ((IWsFederationApplicationSettingsApplication)retrievedSettings.App).GroupValueFormat.Should().Be("windowsDomainQualifiedName");
+                ((IWsFederationApplicationSettingsApplication)retrievedSettings.App).Realm.Should().Be("urn:example:app");
+                ((IWsFederationApplicationSettingsApplication)retrievedSettings.App).WReplyUrl.Should().Be("https://example.com");
+                ((IWsFederationApplicationSettingsApplication)retrievedSettings.App).AttributeStatements.Should().BeNull();
+                ((IWsFederationApplicationSettingsApplication)retrievedSettings.App).NameIdFormat.Should().Be("urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified");
+                ((IWsFederationApplicationSettingsApplication)retrievedSettings.App).AuthnContextClassRef.Should().Be("urn:oasis:names:tc:SAML:2.0:ac:classes:PasswordProtectedTransport");
+                ((IWsFederationApplicationSettingsApplication)retrievedSettings.App).SiteUrl.Should().Be("https://example.com");
+                ((IWsFederationApplicationSettingsApplication)retrievedSettings.App).WReplyOverride.Should().BeFalse();
+                ((IWsFederationApplicationSettingsApplication)retrievedSettings.App).GroupFilter.Should().BeNull();
+                ((IWsFederationApplicationSettingsApplication)retrievedSettings.App).UsernameAttribute.Should().Be("username");
+            }
+            finally
+            {
+                await client.Applications.DeactivateApplicationAsync(createdApp.Id);
+                await client.Applications.DeleteApplicationAsync(createdApp.Id);
+            }
+        }
+
+        /// <summary>
+        /// FAIL: Invalid grantTypes & responseTypes
+        /// </summary>
+        [Fact]
+        public async Task AddOAuthTwoApp()
+        {
+            var client = GetClient();
+
+            var createdApp = await client.Applications.CreateApplicationAsync(new OpenIdConnectApplication
+            {
+                Name = "oidc_client",
+                Label = "Sample Client",
+                SignOnMode = ApplicationSignOnMode.OpenidConnect,
+
+                Credentials = new OAuthApplicationCredentials()
+                {
+                    OauthClient = new ApplicationCredentialsOAuthClient()
+                    {
+                        ClientId = "0oa1hm4POxgJM6CPu0g4",
+                        TokenEndpointAuthMethod = OAuthEndpointAuthenticationMethod.ClientSecretPost,
+                        AutoKeyRotation = true,
+                    },
+                },
+
+                // MISSING PROPERTY: "post_logout_redirect_uris": [ "https://example.com/oauth2/postLogoutRedirectUri"],
```

Comment:
Reminder to discuss ⬆️

Reply:
https://github.com/okta/okta-sdk-dotnet/issues/209